javascript_ts/11.qnamaker/src/index.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { BotFrameworkAdapter, MemoryStorage, ConversationState } from 'botbuilder';
+import { QnAMakerEndpoint } from 'botbuilder-ai';
+import { BotConfiguration, IEndpointService, QnaMakerService, IQnAService, IAppInsightsService } from 'botframework-config';
+import { config } from 'dotenv';
+import * as path from 'path';
+import * as restify from 'restify';
+import { QnAMakerBot } from './bot';
+import { MyAppInsightsMiddleware, MyAppInsightsMiddlewareSettings } from './middleware';
+
+const CONFIG_ERROR = 1;
+
+// bot name as defined in .bot file 
+// See https://aka.ms/about-bot-file to learn more about .bot file its use and bot configuration.
+const BOT_CONFIGURATION = 'qnamaker';
+const QNA_CONFIGURATION = 'qnamakerService';
+const APP_INSIGHTS_CONFIGURATION = 'appInsights';
+
+// Read botFilePath and botFileSecret from .env file
+// Note: Ensure you have a .env file and include botFilePath and botFileSecret.
+const ENV_FILE = path.join(__dirname, '../.env');
+const env = require('dotenv').config({path: ENV_FILE});
+
+// Create HTTP server
+let server = restify.createServer();
+server.listen(process.env.port || process.env.PORT || 3978, function () {
+    console.log(`\n${server.name} listening to ${server.url}`);
+    console.log(`\nGet Bot Framework Emulator: https://aka.ms/botframework-emulator`);
+    console.log(`\nTo talk to your bot, open qnamaker.bot file in the emulator.`);
+});
+
+// .bot file path
+const BOT_FILE = path.join(__dirname, (process.env.botFilePath || ''));
+
+let botConfig: BotConfiguration;
+try {
+    // Read bot configuration from .bot file. 
+    botConfig = BotConfiguration.loadSync(BOT_FILE, process.env.botFileSecret);
+} catch (err) {
+    console.log(`Error reading bot file. Please ensure you have valid botFilePath and botFileSecret set for your environment`);
+    process.exit(CONFIG_ERROR);
+}
+
+// Get bot endpoint configuration by service name.
+const endpointConfig = <IEndpointService>botConfig.findServiceByNameOrId(BOT_CONFIGURATION);
+const qnaConfig = <IQnAService>botConfig.findServiceByNameOrId(QNA_CONFIGURATION);
+const appInsightsConfig = <IAppInsightsService>botConfig.findServiceByNameOrId(APP_INSIGHTS_CONFIGURATION);
```

Comment:
Per discussion with Dave, we should move app insights to its own sample. We should remove this from the QnA sample.